owncloudApp/src/main/java/com/owncloud/android/files/services/FileUploader.java

```diff
@@ -290,8 +289,7 @@ public int onStartCommand(Intent intent, int flags, int startId) {
                 KEY_REQUESTED_FROM_WIFI_BACK_EVENT, false
         );
 
-        if ((isCameraUploadFile || isAvailableOfflineFile || isRequestedFromWifiBackEvent) &&
-                Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
This check was included to fix a problem with notifications here: https://github.com/owncloud/android/issues/2453#issuecomment-466322335 . But if everything is working, we can move this forward @jesmrec

Reply:
If we'd remove that line, the problem you mention will arise again, causing a regression. We must avoid this. CC @hannesa2

Reply:
@abelgardep , @jesmrec This if block is just to

 `startForeground(141, mNotificationBuilder.build());`

nothing else. And because of this if condition it was crashing. This `startForeground()` needs nothing else 

`if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {`

Reply:
@hannesa2 

We were checking `isCameraUploadFile || isAvailableOfflineFile || isRequestedFromWifiBackEvent` conditions to execute `startForeground(141, mNotificationBuilder.build())` because it needs to be called within five seconds after [calling `startForegroundService` from TransferRequester](https://github.com/owncloud/android/blob/master/owncloudApp/src/main/java/com/owncloud/android/files/services/TransferRequester.java#L95) for those cases (cameraUploads, available offline and requestdFromWifiBackEvent), otherwise, the app will crash. 

But you're also right, we should call `startForeground()` after [calling startService() from TransferRequester](https://github.com/owncloud/android/blob/master/owncloudApp/src/main/java/com/owncloud/android/files/services/TransferRequester.java#L98) to make FileUploader service run in foreground. Otherwise, the app could fail if the system decides to kill FileUploader (it does not usually happen). Could you tell me in which scenario is the previous code crashing for you?

What really concerns me most is the duplicated notification that appears when trying your solution, and we had to fix in https://github.com/owncloud/android/issues/2453#issuecomment-466322335 . Besides, we are also able to show a notification even when calling `startService() ` in a normal upload. It seems that many changes have happened here in the Android code itself.

Could you have a look at the [problem with duplicated notifications](https://github.com/owncloud/android/issues/2453#issuecomment-466322335) that seems to happen after applying your changes? Thanks